core/src/main/java/cucumber/runtime/arquillian/feature/FeaturesManager.java

```diff
@@ -0,0 +1,370 @@
+    }
+    
+    private static Collection<String> findFeatures(final Class<?> javaClass) {
+        final Collection<String> featureUrls = new ArrayList<String>();
+        { // convention
+            final String featurePath = getfeaturePath(javaClass);
+            featureUrls.add(featurePath);
+        }
+        { // our API
+            final Features additionalFeaturesAnnotation = javaClass.getAnnotation(Features.class);
+            if (additionalFeaturesAnnotation != null) {
+                Collections.addAll(featureUrls, additionalFeaturesAnnotation.value());
+            }
+        }        
+        { // cucumber-junit API
+            final CucumberOptions cucumberOptionsAnnotation = javaClass.getAnnotation(CucumberOptions.class);
+            if (cucumberOptionsAnnotation != null && cucumberOptionsAnnotation.features() != null) {
+                Collections.addAll(featureUrls, cucumberOptionsAnnotation.features());
+            }
+        }
+        return featureUrls;
+    }
+    
+    private static String getfeaturePath(final Class<?> javaClass) {
+        return javaClass.getPackage().getName().replace('.', '/') + '/' + createClassNameSubPackage(javaClass.getSimpleName()) + EXTENSION;
+    }
+    
+    private static String createClassNameSubPackage(final String name) {
+        String result = name;
+        if (result.endsWith("Test")) {
+            result = result.substring(0, result.length() - "Test".length());
+        } else if (result.endsWith("IT")) {
+            result = result.substring(0, result.length() - "IT".length());
+        }
+
+        if (result.length() == 1) {
+            return result;
+        }
+        return Character.toLowerCase(result.charAt(0)) + replaceUpperCaseWithADashAndLowercase(result.substring(1));
+    }
+    
+    private static String replaceUpperCaseWithADashAndLowercase(final String substring) {
+        final StringBuilder builder = new StringBuilder();
+        for (final char c : substring.toCharArray()) {
+            if (!Character.isUpperCase(c)) {
+                builder.append(c);
+            } else {
+                builder.append('-').append(Character.toLowerCase(c));
+            }
+        }
+
+        final String s = builder.toString();
+        if (s.endsWith("-")) {
+            return s.substring(0, s.length() - 1);
+        }
+        return s;
+    }
+    
+    private static Map<String, Collection<URL>> extractUrlMapFromRawFeature(final String rawFeatureURL, final String featureHome, final String tempDir, 
+    		final Collection<ResourceLoader> customResourceLoaders,final ClassLoader classLoader, final Class<?> javaClass) {
+
+    	final Map<String, Collection<URL>> featureURLMap = new HashMap<String, Collection<URL>>();
+        final String urlPath = extractFeatureURLPath(getSeparatorIndex(rawFeatureURL), rawFeatureURL);
+        final boolean directResource = urlPath.endsWith(EXTENSION);
+        
+        if (directResource) {
```

Comment:
I cut the method declaration in two lines because of the parameters' list, found it a bit long to be kept singlelined